cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -306,6 +317,7 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 			p.unremovableNodes.AddTimeout(unremovable, unremovableTimeout)
 		}
 	}
+	p.handleUnprocessedNodes(nil, time.Now(), &endedPrematurely)
```

Comment:
nit: I'm not a fan of passing `endedPrematurely` by pointer and using it as a both input and output of this function. Alternative to consider:

```
var skippedNodes []string
(...)
  if timedOut(timer) {
     skippedNodes = currentlyUnneededNodeNames[i:]
     (...)
  }
  (...)
}
p.handleUnprocessedNodes(skippedNodes) // or just inline
(...)
```